ironfist/src/cpp/scripting/scripting.cpp

```diff
@@ -772,41 +772,163 @@ char *GetScriptContents() {
   return script_contents;
 }
 
-void GetNextMapVariable(int &key, const char *&mapVariableId, const char *&mapVariableValue) {
-  // PUSH THE MAP VARIABLES LIST TO THE TOP OF THE STACK OR nil IF THERE IS NO LIST
-  lua_getglobal(map_lua, "mapVariables");
-
-  // CHECK IF THERE IS INDEED A LIST OF MAP VARIABLES
-  if (lua_isnil(map_lua, -1)) {
-    mapVariableId = "noMapVariables";
-    return;
-  }
-
-  // GO TO THE mapVariable INDEXED BY key
-  if (key >= 0) {
-    lua_pushnumber(map_lua, key);
-  }
-  else {
-    lua_pushnil(map_lua);
-  }
-  if (lua_next(map_lua, -2)) {
-    key = lua_tointeger(map_lua, -2);
-    mapVariableId = lua_tostring(map_lua, -1);
-    lua_pop(map_lua, 2);
-    lua_getglobal(map_lua, mapVariableId);
-    mapVariableValue = lua_tostring(map_lua, -1);
-  }
-  else {
-    mapVariableId = "noMapVariables";
-  }
-}
-
-void SetMapVariables(const char *id, const char *quantity) {
-  lua_pushstring(map_lua, quantity);
-  lua_setglobal(map_lua, id);
+void ErrorMapVariable(std::string& mapVariableId, const std::string& s2, const std::string& addErrorMessage) {
+	const std::string s1("MapVariable '");
+	const std::string errorMessage = s1 + mapVariableId + s2 + addErrorMessage;
+	const std::string errorLabel("mapVariable Error");
+	DisplayError(errorMessage, errorLabel);
+}
+
+void ErrorSavingMapVariable(std::string& mapVariableId, const std::string& addErrorMessage) {
+	const std::string s2("' could not be saved properly.");
+	ErrorMapVariable(mapVariableId, s2, addErrorMessage);
+}
+
+void ErrorLoadingMapVariable(std::string& mapVariableId, const std::string& addErrorMessage) {

```

Comment:
Hmm. My suspicion is that my screen width was such that the brace got displayed on the next line, and I didn't realize it was actually on the same line. This is fine.